src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -237,7 +237,9 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             // not function properly.
             Href = output.Attributes[HrefAttributeName]?.Value as string;
 
-            var modeResult = AttributeMatcher.DetermineMode(context, ModeDetails);
+            var determinePartiallyMatchedAttributes = Logger.IsEnabled(
+                ModeMatchResultLoggerExtensions.PartiallyMatchedAttributesLogLevel);
+            var modeResult = AttributeMatcher.DetermineMode(context, ModeDetails, determinePartiallyMatchedAttributes);
             Logger.TagHelperModeMatchResult(modeResult, context.UniqueId, ViewContext.View.Path, this);
```

Comment:
The `Logger` call? That was intended to aid developers when debugging why a Tag Helper didn't run the way they expected, e.g. missing an argument. Without it, it's pretty hard to know why a Tag Helper didn't "work".